backend/windmill-queue/src/queue_transaction.rs

```diff
@@ -0,0 +1,161 @@
+use std::fmt::Debug;
+
+use futures_core::{future::BoxFuture, stream::BoxStream};
+use rsmq_async::{RedisBytes, RsmqConnection};
+use sqlx::{Postgres, Transaction};
+
+pub enum RedisOp {
+    SendMessage(RedisBytes, Option<chrono::DateTime<chrono::Utc>>),
+    DeleteMessage(String),
+}
+
+impl RedisOp {
+    pub async fn apply<R: RsmqConnection>(self, rsmq: &mut R) -> Result<(), rsmq_async::RsmqError> {
+        match self {
+            RedisOp::SendMessage(bytes, time) => {
+                rsmq.send_message(
+                    "main_queue",
+                    bytes,
+                    time.map(|t| (t - chrono::Utc::now()).num_seconds())
```

Comment:
this should use the database now() otherwise there are risks of drift since the clock of the worker might be the same as the db